test/integration_test/action_test.go

```diff
@@ -0,0 +1,223 @@
+//go:build integrationtest
+// +build integrationtest
+
+package integrationtest
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"testing"
+	"time"
+
+	"github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/portworx/sched-ops/k8s/core"
+	storkops "github.com/portworx/sched-ops/k8s/stork"
+	"github.com/portworx/sched-ops/task"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAction(t *testing.T) {
+	// Create secrets on source and destination
+	// Since the secrets need to be created on the destination before migration
+	// is triggered using the API instead of spec factory in torpedo
+	secret := createSecret()
+	setSecret(t, secret, "src")
+	setSecret(t, secret, "dest")
+
+	t.Run("actionFailoverTest", actionFailoverTest)
+}
+
+func createSecret() *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      "volume-secrets",
+			Namespace: "kube-system",
+		},
+		StringData: map[string]string{
+			"mysql-secret": "supersecretpassphrase",
+		},
+	}
+}
+
+func setSecret(t *testing.T, secret *v1.Secret, cluster string) {
+	if cluster == "dest" {
+		logrus.Infof("creating secret in destination")
+		err := setDestinationKubeConfig()
+		require.NoError(t, err, "failed to set kubeconfig to destination cluster: %v", err)
+
+		defer func() {
+			err := setSourceKubeConfig()
+			require.NoError(t, err, "failed to set kubeconfig to source cluster: %v", err)
+		}()
+	} else {
+		logrus.Infof("creating secret in source")
+	}
+
+	_, err := core.Instance().CreateSecret(secret)
+	if !errors.IsAlreadyExists(err) {
+		require.NoError(t, err, "failed to create secret for volumes")
```

Comment:
```suggestion
		require.NoError(t, err, "failed to create secret on %s for volumes", cluster)
```

Reply:
can you also please address the above suggestion, if the test fails, it will help us know faster on which cluster secret creation failed.